virtio-devices/src/vsock/unix/muxer.rs

```diff
@@ -1454,4 +1500,123 @@ mod tests {
+    fn test_parse_command() {
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("")),
+            Err(Error::UnixRead(_))
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("\n")),
+            Err(Error::InvalidPortRequest)
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("CONN\n")),
+            Err(Error::InvalidPortRequest)
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("FOO ")),
+            Err(Error::UnixRead(_))
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("FOOFOOX ")),
+            Err(Error::InvalidPortRequest)
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("CONNECT ")),
+            Err(Error::UnixRead(_))
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("connect ")),
+            Err(Error::UnixRead(_))
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str("connect \n")),
+            Err(Error::InvalidPortRequest)
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str(
+                "connect 1337"
+            )),
+            Err(Error::UnixRead(_))
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str(
+                "connect -1337\n"
+            )),
+            Err(Error::InvalidPortRequest)
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str(
+                "connect 8589934592\n"
+            )),
+            Err(Error::InvalidPortRequest)
+        ));
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str(
+                "CONNECT 👾\n"
+            )),
+            Err(Error::InvalidPortRequest)
+        ));
+        let max_buf_length_no_newline = "CONNECT                        1";
+        assert_eq!(
+            max_buf_length_no_newline.len(),
+            PARTIALLY_READ_COMMAND_BUF_SIZE
+        );
+        assert!(matches!(
+            VsockMuxer::parse_port_from_read_command(&PartiallyReadCommand::from_str(
+                max_buf_length_no_newline
+            )),
+            Err(Error::InvalidPortRequest)
+        ));
+        let max_buf_length_correct = "CONNECT                       1\n";
+        assert_eq!(
+            max_buf_length_correct.len(),
+            PARTIALLY_READ_COMMAND_BUF_SIZE
+        );
+        assert!(matches!(
```

Comment:
This could be `assert_eq!`.

Reply:
It cannot (simply) as `PartialEq` needs to be implemented for `Error` which has a variant that embeds `std::io::Error` which does not implement `PartialEq`.
I would only work with e.g. `unwrap`

Reply:
Ah, right.